Money_Counter_JB.py

```diff
@@ -0,0 +1,44 @@
+# This programme counts British coins based on their weight
+
+# Started with dictionary of UK coin weights
+
+coindict = {
+    "1": 3.56,
+    "2": 7.12,
+    "5": 3.25,
+    "10": 6.5,
+    "20": 5.0,
+    "50": 8.0
+    }
+
+while True:     # Put this eternal loop in so I don't have to keep restarting it
+
+    print()     # Blank lines for aesthetics
+
+    objective = input("\nWould you like to \n\n(a) weigh coins \nor \n(b) fill a coin bag?\n\n " ) # First choice: find the value of a batch of coins or fill a bag correctly?
+        
+    if objective == ("b"): # The fill a bag option
+        
+        bag=int(input("\n\nHow much does the bag take? In £: ")) # ASss for fill value printed on the bag
+        
+        choice = input("\n\nWhat coin denomination do you have in pence? (i.e. 20): ") # Asks for coin type
+        singleweight = coindict.get(choice) # Used the key to set variable of coin weight using dictionary values
+        filled = bag*100 # Got in a pickle trying to do this all in the line below so took it out and made a new variable
+        weight = (((1 / int(choice)) * (singleweight))*filled) # The answer of how much weight of given coins fill a bag
+        
+        print("\n\nYou'll need " + str(weight) + " grams of " + (str(choice)) + "p pieces") # Answers in a sentence
```

Comment:
You can utilise f-strings here again instead of concatenation like you've done. Concatenation is great, but f-strings are a string-specific syntax, and this is less verbose, and we just love them. No need to cast the integer to a string either.

```suggestion
print(
        f"\n\nYou'll need {weight}  grams of {choice}p pieces"
 ) # Answers in a sentence
```